src/ln/channel.rs

```diff
@@ -1400,6 +1421,27 @@ impl Channel {
 			return Err(ChannelError::Close("We consider the minimum depth to be unreasonably large"));
 		}
 
+		let their_shutdown_scriptpubkey = if their_local_features.supports_upfront_shutdown_script() {
+			match &msg.shutdown_scriptpubkey {
+				&OptionalField::Present(ref script) => {
+					// Peer is signaling upfront_shutdown and has provided a non-accepted scriptpubkey format. We enforce it while receiving shutdown msg
+					if script.is_p2pkh() || script.is_p2sh() || script.is_v0_p2wsh() || script.is_v0_p2wpkh() || script.len() == 0 {
```

Comment:
the script.len()== 0 check at the end breaks the next if.

Reply:
Are you on the latest tip ? I've pushed a version with this bug but it was corrected at least on b603d94

Reply:
Looks like its still there on the latest?

Reply:
Sorry, I've forgot to correct on both open_channel/accept_channel case, extend test_option_upfront_shutdown_script to cover it